src/gadgets/num.rs

```diff
@@ -8,6 +8,9 @@ use super::Assignment;
 
 use super::boolean::{self, AllocatedBit, Boolean};
 
+use serde::{Deserialize, Serialize};
```

Comment:
This import should be right below (without an empty line) line 3, the `use ff` import.